Sources/Kitura/CodableRouter+TypedMiddleware.swift

```diff
@@ -0,0 +1,1020 @@
+/*
+ * Copyright IBM Corporation 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import LoggerAPI
+import KituraNet
+import KituraContracts
+import Dispatch
+
+// Codable router
+
+extension Router {
+
+    // Used by PUT and PATCH with identifier
+    public typealias MiddlewareIdentifierCodableClosure<T: TypeSafeMiddleware, Id: Identifier, I: Codable, O: Codable> = (T, Id, I, @escaping CodableResultClosure<O>) -> Void
```

Comment:
Is there a reason why these typealiases are inside `extension Router {}`? I would have thought they should be top-level.